examples/flex-flows/environment-variable-connection-mapping/README.md

```diff
@@ -0,0 +1,112 @@
+# How to use environment variable with value from connection in prompt flow
+When developer write an LLM app from scratch, they may not use the prompt flow connection. Instead, It's common to use environment variables to set API keys in local. In this example, we provide a tutorial about how to set environment variable with the values from connection. We call this as EVC (environment variable with connection) mapping.
+
+## Prerequisites
+
+Install promptflow sdk and other dependencies in this folder:
+
+## What you will learn
+Upon completing this tutorial, you should be able to:
+
+- Learn how to set environment variable in prompt flow.
+- Learn how to leverage prompt flow to load the secret values in connection into environment variable.
+- Know the scenarios support using this feature.
+
+We assume you've tried some other flex flow samples before learning this one. You know how to do flow test and batch run with a flex flow.
+
+## Invoke flex flow using python intepreter
+
+When you build your LLM from scratch, the environment variable comes from your own source. You can use below ways to set environment variables 
+- Setup environment variables via terminal
+bash:
+```bash
+export AZURE_OPENAI_API_KEY="your_key_here"
+export AZURE_OPENAI_ENDPOINT="your_endpoint"
+```
+powershell:
+```ps
+$env:AZURE_OPENAI_API_KEY = "your_key_here"
+$env:AZURE_OPENAI_ENDPOINT = "your_endpoint"
+```
+command line:
+```cmd
+set AZURE_OPENAI_API_KEY="your_key_here"
+set AZURE_OPENAI_ENDPOINT="your_endpoint"
+```
+
+- Setup environment variables via .env file
+
+Ensure you have put your azure open ai endpoint key in [.env](../.env) file. You can create one refer to this [example file](../.env.example).
+
+```bash
+cat .env
+```
+
+- Test flow
+
+```bash
+# run flow directly
+python ./flow.py
+```
+
+## EVC mapping format
+EVC mapping is a section in yaml/CLI parameter, the format is like below. The section's name is environment_variables, it's a list of key-value pair. The key represent the environment variable name. The value can be a string value of the environment variable, or a conneciton'sfield wrap with ${}. 
+*To avoid expose plain secret in anywhere, we strongly recommend you to always use the connection field as the value in EVC mapping, instead of put real keys into this section. *
+```bash
+environment_variables: 
+  AZURE_OPENAI_API_KEY: ${some_connection.api_key} 
```

Comment:
may need to educate user how to create connection? 
And we can provide cli to  create a AOAIConnection from .env file or environment variables?

Reply:
Something like? 
pf connection create -f .env --name openai --type open_ai # .env: from dotenv file
pf connection create -f env --name openai --type open_ai # env: special name for create from env

Reply:
@brynn-code

Reply:
"may need to educate user how to create connection?" -- added a refer link to connection sampes

Reply:
"And we can provide cli to create a AOAIConnection from .env file or environment variables?" similar to the from_env function we added recently?